scripts/mbedtls_dev/bignum_mod.py

```diff
@@ -30,6 +30,26 @@ class BignumModTarget(test_data_generation.BaseTarget):
 
 # BEGIN MERGE SLOT 2
 
+class BignumModMul(bignum_common.ModOperationCommon,
+                   BignumModTarget):
+    # pylint:disable=duplicate-code
+    """Test cases for bignum mpi_mod_mul()."""
+    symbol = "*"
+    test_function = "mpi_mod_mul"
+    test_name = "mbedtls_mpi_mod_mul"
+    input_style = "arch_split"
+    arity = 2
+
+    def arguments(self) -> List[str]:
+        return [self.format_result(self.to_montgomery(self.int_a)),
+                self.format_result(self.to_montgomery(self.int_b)),
+                bignum_common.quote_str(self.arg_n)
+               ] + self.result()
```

Comment:
This section is still causing pylint duplicate code issues unfortunately. Ideally we would use a better solution for this like deriving from a common class with the shared methods, but that is existing tech debt. A simpler option for this PR would be use an equivalent implementation which is different enough to make pylint happy, for example:
```suggestion
        return [self.format_result(n) for n in [self.to_montgomery(self.int_a),
                                                self.to_montgomery(self.int_b),
                                                self.int_n]
               ] + self.result()
```

Reply:
This could also use the recently-added `montgomery_form_a` (and add its own `montgomery_form_b`?)  With that, and switching to the "N-first-for-tests" form, this function wouldn't be needed at all, IIUC

Reply:
I still wouldn't fix the underlying issue which is pylint forcing us to workaround the design, and write code which does the same thing in different ways. Updating this forked logic is a mess. 

As for what this method is doing, how does the output compare to what the super().arguments() returns when called from the test case? Because it looks to me as we we want the standard return, but choose to put modulo in the end, and convert a, b to montgomery. Is that the intent?

`ModOperationCommon.arguments()`

~~~~~~
    def arguments(self) -> List[str]:
        return [quote_str(self.arg_n)] + super().arguments()
~~~~~~

The way I would do it is provide discrete methods to return arguments in the standard by design format ( N, A, B, result)

~~~~~
    def arguments_int(self) -> List[str]:
        return [self.arg_n, 
                    self.int_a,
                    self.int_b] if cls.arity == 2 else [self.arg_n,
                                                                      self.int_a]

    def arguments_mont(self) -> List[str]:
        return [self.arg_n, 
                    self.to_montgomery(self.int_a),
                    self.to_montgomery(self.int_a)] if cls.arity == 2 else [self.arg_n,
                                                                                                    self.int_a]
~~~~~

Since most of the tests will need to use one of those.

The question is weather pylint will complain when you do the obvious one liner

~~~~~
def arguments(self) -> List[str]:
     return [self.format_result(n) for n in super.arguments_mont() + + self.result()]
~~~~~